mediapipe/tasks/c/vision/pose_landmarker/pose_landmarker_test.cc

```diff
@@ -0,0 +1,262 @@
+              landmark_precision);
+  EXPECT_NEAR(result->pose_world_landmarks[0].landmarks[0].z, -0.1469f,
+              landmark_precision);
+}
+
+TEST(PoseLandmarkerTest, ImageModeTest) {
+  const auto image = DecodeImageFromFile(GetFullPath(kImageFile));
+  ASSERT_TRUE(image.ok());
+
+  const std::string model_path = GetFullPath(kModelName);
+  PoseLandmarkerOptions options = {
+      /* base_options= */ {/* model_asset_buffer= */ nullptr,
+                           /* model_asset_buffer_count= */ 0,
+                           /* model_asset_path= */ model_path.c_str()},
+      /* running_mode= */ RunningMode::IMAGE,
+      /* num_poses= */ 1,
+      /* min_pose_detection_confidence= */ 0.5,
+      /* min_pose_presence_confidence= */ 0.5,
+      /* min_tracking_confidence= */ 0.5,
+      /* output_segmentation_masks= */ true,
+  };
+
+  void* landmarker = pose_landmarker_create(&options, /* error_msg */ nullptr);
+  EXPECT_NE(landmarker, nullptr);
+
+  const auto& image_frame = image->GetImageFrameSharedPtr();
+  const MpImage mp_image = {
+      .type = MpImage::IMAGE_FRAME,
+      .image_frame = {.format = static_cast<ImageFormat>(image_frame->Format()),
+                      .image_buffer = image_frame->PixelData(),
+                      .width = image_frame->Width(),
+                      .height = image_frame->Height()}};
+
+  PoseLandmarkerResult result;
+  pose_landmarker_detect_image(landmarker, mp_image, &result,
+                               /* error_msg */ nullptr);
+  MatchesPoseLandmarkerResult(&result, kLandmarkPrecision);
+  pose_landmarker_close_result(&result);
+  pose_landmarker_close(landmarker, /* error_msg */ nullptr);
+}
+
+TEST(PoseLandmarkerTest, VideoModeTest) {
+  const auto image = DecodeImageFromFile(GetFullPath(kImageFile));
+  ASSERT_TRUE(image.ok());
+
+  const std::string model_path = GetFullPath(kModelName);
+  PoseLandmarkerOptions options = {
+      /* base_options= */ {/* model_asset_buffer= */ nullptr,
+                           /* model_asset_buffer_count= */ 0,
+                           /* model_asset_path= */ model_path.c_str()},
+      /* running_mode= */ RunningMode::VIDEO,
+      /* num_poses= */ 1,
+      /* min_pose_detection_confidence= */ 0.5,
+      /* min_pose_presence_confidence= */ 0.5,
+      /* min_tracking_confidence= */ 0.5,
+      /* output_segmentation_masks= */ true,
+  };
+
+  void* landmarker = pose_landmarker_create(&options, /* error_msg */ nullptr);
+  EXPECT_NE(landmarker, nullptr);
+
+  const auto& image_frame = image->GetImageFrameSharedPtr();
+  const MpImage mp_image = {
+      .type = MpImage::IMAGE_FRAME,
+      .image_frame = {.format = static_cast<ImageFormat>(image_frame->Format()),
+                      .image_buffer = image_frame->PixelData(),
+                      .width = image_frame->Width(),
+                      .height = image_frame->Height()}};
+
+  for (int i = 0; i < kIterations; ++i) {
```

Comment:
I don't think we gain much by running this more than 2 times. What do you think?

(The live stream test is different since it tests flow limits).

Reply:
The C++ test does it so thought I'd do it here too.

https://github.com/google/mediapipe/blob/495012645626e1b1e4da98a67919c53ac7f1520f/mediapipe/tasks/cc/vision/pose_landmarker/pose_landmarker_test.cc#L386

We can cut it down since we already know it runs well in C++ perhaps?

Reply:
Yeah - that's what I am thinking. Will set to 10 during import.

Reply:
( I see you already set it to 5 - that's good as well. So keeping this).